crates/context_aware_config/src/helpers.rs

```diff
@@ -195,6 +207,44 @@ pub fn calculate_context_priority(
     }
 }
 
+fn calculate_weight_from_index(index: u32) -> Result<BigDecimal, String> {
+    let base = BigUint::from(2u32);
+    let result = base.pow(index);
+    let biguint_str = &result.to_str_radix(10);
```

Comment:
`let biguint_str = &result.to_str_radix(10);` and `BigDecimal::from_str_radix(&biguint_str, 10)` seem complementary, we could directly use `result` ?

what is the format of `result`, is it not in some desired format, because of which we need to to these conversions ?

Reply:
result is of type BigUint
from_str_radix returns type Bigdecimal